### PR TITLE
Expanded MNIST example with colab notebooks

### PR DESCRIPTION
**Pull request by jimthompson5802**
Updated MNIST example for Ludwig 0.5

### CODE REVIEW

**Comment by w4nderlust** [comment]
Looks great to me! I think there's a little issue with pre-commit, maybe @justinxzhao chan help with that, but other than that this looks great.

**Comment by justinxzhao** [comment]
Just merged in a fix for the pre-commits. Should be green if you rerun the check now

File: docs/examples/mnist.md [hunk 1]
```diff
@@ -1,114 +1,239 @@
...
-tar -xf mnist_png.tar.gz
-cd mnist_png/
+[MNIST](http://yann.lecun.com/exdb/mnist/) is a collection of gray-scale images of handwritten digits. This collection is made up of 60,000 images for training and 10,000 images for testing model performance.  Each image is 28 X 28 pixels in gray-scale.
+
+```shell
+ludwig datasets download mnist
 ```
 
-## Create train and test CSVs
+This command will create a dataset `mnist_dataset.csv` in the current directory.
```
Comment (dantreiman) [inline docs/examples/mnist.md:19]: Maybe also mention that `ludwig datasets download mnist` creates `training/` and `testing/` directories in the current dir with the dataset images.

File: docs/examples/mnist.md [hunk 2]
```diff
@@ -1,114 +1,239 @@
...
-Now you should have `mnist_dataset_training.csv` and `mnist_dataset_testing.csv`
-containing 60000 and 10000 examples correspondingly and having the following format
+## Train
 
-| image_path           | label |
-| -------------------- | ----- |
-| training/0/16585.png | 0     |
-| training/0/24537.png | 0     |
-| training/0/25629.png | 0     |
+The Ludwig configuration file describes the machine learning task.  There is a vast array of options to control the learning process.  This example only covers a small fraction of the options.  Only the options used in this example are described.  Please refer to the [Configuration Section](../../configuration) for all the details.
```
Comment (dantreiman) [inline docs/examples/mnist.md:34]: Condensed wording.  Just a suggestion, not really necessary.
```suggestion
The Ludwig configuration file describes the machine learning task. This example only uses a small subset of the options provided by Ludwig. Please refer to the [Configuration Section](../../configuration) for all the details.
```

**Review by dantreiman — APPROVED** [review]
Looks great to me.  Thanks!  Left a few minor comments.

File: docs/examples/mnist.md (hunk 1 above)
Comment (jimthompson5802) [reply docs/examples/mnist.md:19]: done

File: docs/examples/mnist.md [hunk 3]
```diff
@@ -1,114 +1,239 @@
...
+      --config config.yaml
+    ```
+
+=== "python"
+
+    [train() method](../../user_guide/api/LudwigModel/#train)
+
+    ```python
+    # Trains the model. This cell might take a few minutes.
+    train_stats, preprocessed_data, output_directory = model.train(dataset=train_df)
```
[the remark this answers is not in the document]
Comment (jimthompson5802) [reply docs/examples/mnist.md:137]: added write-up for  api download

**Comment by jimthompson5802** [comment]
Migrated this source branch to `ludwig-ai/ludwig-docs` repo.  work continues from that repo

**Pull request closed without merging**